File: all-submissions/0034-find-first-and-last-position-of-element-in-sorted-array/solution.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int left =0;
        int right=nums.length-1;
        int startIndex = -1;
        int endIndex=-1;
        if(nums.length == 1) {
            if (target == nums[0]) {
                return new int[]{0,0};
            }
        }

        while(left<=right) {
            int mid = (left+right)/2;
            if(nums[mid] < target) {
                left = mid+1;
            } 
            else if(nums[mid] > target) {
                right = mid-1;
            } else if(nums[mid] == target) {
                int tempMid = mid;
                startIndex = mid;
                endIndex = mid;
                while(mid>=left){
                    if(nums[mid] == target) {
                        startIndex = mid;
                        mid--;
                    }
                    else{
                        break;
                    }
                }
                mid = tempMid;
                while(mid<=right){
                    if(nums[mid] == target) {
                        endIndex = mid;
                        mid++;
                    }
                    else{
                        break;
                    }
                }
                break;
            }
        }
        return new int[]{startIndex, endIndex};

        
    }
}

File: all-submissions/0034-find-first-and-last-position-of-element-in-sorted-array/solution.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        
        
        def findLeftMostIndex():
            left = 0
            right = len(nums)-1
            while left < right:
                mid = (left + right) // 2
                if nums[mid] >= target:
                    right = mid
                else:
                    left = mid + 1

            return left if 0<=left<len(nums) and nums[left] == target else -1

        def findRightMostIndex():
            left = 0
            right = len(nums)-1
            while left <= right:
                mid = (left + right) // 2
                if nums[mid] > target:
                    right = mid-1
                else:
                    left = mid + 1

            return right if 0<=right<len(nums)  and nums[right] == target else -1

        l =findLeftMostIndex()
        r = findRightMostIndex()
        return [l,r]
